#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.9999157660218978 0.012935555742376188 -0.0010640763492404 5.349986391620014
  position -0.049677326457612 2.838869356071242 1.9768832613293499
}
TexturedBackground {
  texture "factory"
}
TexturedBackgroundLight {
}
WoodenBox {
  translation -0.3841 -4.44089e-16 0.0584928
  name "wooden box(1)"
  size 0.251 0.2 0.01
}
WoodenBox {
  translation -0.512158 6.66134e-16 -0.189547
  name "wooden box(3)"
  size 0.508 0.2 0.01
}
WoodenBox {
  translation -0.00127955 -1.24345e-14 -0.57098
  rotation 0 1 0 -1.5707953071795862
  name "wooden box(2)"
  size 0.251 0.2 0.01
}
WoodenBox {
  translation -0.507275 -1.13243e-14 0.189905
  rotation 0 1 0 -1.5707953071795862
  name "wooden box(4)"
  size 0.251 0.2 0.01
}
WoodenBox {
  translation -0.00641642 -1.11023e-14 -0.064548
  rotation 0 1 0 -1.5707953071795862
  name "wooden box(5)"
  size 0.251 0.2 0.01
}
DEF area RectangleArena {
  translation -0.257628 -4.44089e-16 -0.192172
  floorSize 1.016 1.016
  floorTileSize 0.508 0.508
  wallAppearance BrushedAluminium {
    colorOverride 0 0 0
  }
}
WoodenBox {
  translation -0.259053 -1.24345e-14 -0.325153
  rotation 0 1 0 -1.5707953071795862
  name "wooden box(8)"
  size 0.251 0.2 0.01
}
WoodenBox {
  translation -0.385172 -1.24345e-14 -0.444885
  rotation 0 1 0 1.01503e-06
  name "wooden box(7)"
  size 0.251 0.2 0.01
}
WoodenBox {
  translation 0.124931 2.53129e-14 -0.191002
  rotation 0 1 0 3.14159
  name "wooden box(6)"
  size 0.251 0.2 0.01
}
DEF epuck E-puck {
  translation 0.124246 -9.19265e-14 0.177951
  rotation 0.6981043682969243 0.02240501182014799 0.7156453775509845 -0.0005653071795865827
  controller "path-planning_mapping"
  supervisor TRUE
  camera_width 50
  turretSlot [
    InertialUnit {
      rotation 0 1 0 3.14
      noise 0.1
    }
    DEF righthinge HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          consumptionFactor 70
          controlPID 8 2 0.5
        }
        PositionSensor {
          name "right wheel sensor"
          noise 0.1
        }
      ]
      endPoint Solid {
        translation 0.045009957783800944 0.025013380758883635 -2.467192985207017e-05
        rotation 0.9999993203499804 0.0010052734270078168 -0.0005905293508524927 4.6368371220546685
        children [
          DEF WHEEL Transform {
            translation -0.007 0.01 -0.05
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0 0 0
                  transparency 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
            rotationStep 0.25
          }
        ]
        name "right wheel(1)"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
        rotationStep 0.25
      }
    }
    DEF lefthinge HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          consumptionFactor 70
          controlPID 8 0 0
        }
        PositionSensor {
          name "left wheel sensor"
          noise 0.1
        }
      ]
      endPoint Solid {
        translation -0.04501012942703836 0.024977671977279067 -4.573397616203503e-05
        rotation -0.9999982113395169 0.00047275268613733136 -0.0018313444964861221 1.242581020288959
        children [
          DEF WHEEL Transform {
            translation 0.007 0 -0.05
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.1 0.1 0.1
                  transparency 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
            rotationStep 0.25
          }
        ]
        name "left wheel(1)"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    DEF front_ds DistanceSensor {
      translation 0 0 -0.045
      rotation 0 1 0 1.5708
      name "front_ds"
      recognitionColors [
        0 0 0
      ]
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF right_ds DistanceSensor {
      translation 0.035 -0.01 -0.01
      name "right_ds"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    DEF left_ds DistanceSensor {
      translation -0.035 -0.01 -0.01
      rotation 0 1 0 3.14159
      name "left_ds"
      lookupTable [
        0 0 0
        1.27 1.27 0.1
      ]
    }
    Camera {
      translation 0 -0.01 -0.045
      name "camera1"
      recognitionColors [
        1 1 0
      ]
      fieldOfView 1.57
      width 80
      height 80
      recognition Recognition {
      }
    }
  ]
}
